### PR TITLE
Add separator option to select box

### PR DESCRIPTION
**Pull request by mrleemurray**
Introduce a separator option for the select box, enhancing the dropdown's visual organization. Update focus outlines and styling for separators to improve user experience and maintain consistency with the select button's font size.

image

image

image

### CODE REVIEW

File: src/vs/base/browser/ui/selectBox/selectBoxCustom.ts [hunk 1]
```diff
@@ -997,10 +1020,10 @@ export class SelectBoxList extends Disposable implements ISelectBoxDelegate, ILi
 			this.selected = this.selectList.getFocus()[0];
 
 			// Shift selection down if we land on a disabled option
-			if (this.options[this.selected].isDisabled && this.selected < this.options.length - 1) {
+			while (this.selected < this.options.length - 1 && this.options[this.selected].isDisabled) {
 				this.selected++;
-				this.selectList.setFocus([this.selected]);
 			}
+			this.selectList.setFocus([this.selected]);
```
[the remark this answers is not in the document]
Comment (mrleemurray) [reply src/vs/base/browser/ui/selectBox/selectBoxCustom.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/microsoft/vscode/pull/297703#discussion_r2853773616)

File: src/vs/base/browser/ui/selectBox/selectBoxCustom.ts [hunk 2]
```diff
@@ -1016,10 +1039,10 @@ export class SelectBoxList extends Disposable implements ISelectBoxDelegate, ILi
 			this.selected = this.selectList.getFocus()[0];
 
 			// Shift selection up if we land on a disabled option
-			if (this.options[this.selected].isDisabled && this.selected > 0) {
+			while (this.selected > 0 && this.options[this.selected].isDisabled) {
 				this.selected--;
-				this.selectList.setFocus([this.selected]);
 			}
+			this.selectList.setFocus([this.selected]);
```
[the remark this answers is not in the document]
Comment (mrleemurray) [reply src/vs/base/browser/ui/selectBox/selectBoxCustom.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/microsoft/vscode/pull/297703#discussion_r2853773665)

File: src/vs/base/browser/ui/selectBox/selectBoxCustom.ts [hunk 3]
```diff
@@ -1032,7 +1055,7 @@ export class SelectBoxList extends Disposable implements ISelectBoxDelegate, ILi
 			return;
 		}
 		this.selected = 0;
-		if (this.options[this.selected].isDisabled && this.selected > 1) {
+		while (this.selected < this.options.length - 1 && this.options[this.selected].isDisabled) {
 			this.selected++;
 		}
```
[the remark this answers is not in the document]
Comment (mrleemurray) [reply src/vs/base/browser/ui/selectBox/selectBoxCustom.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/microsoft/vscode/pull/297703#discussion_r2853773683)

File: src/vs/base/browser/ui/selectBox/selectBoxCustom.ts [hunk 4]
```diff
@@ -1047,7 +1070,7 @@ export class SelectBoxList extends Disposable implements ISelectBoxDelegate, ILi
 			return;
 		}
 		this.selected = this.options.length - 1;
-		if (this.options[this.selected].isDisabled && this.selected > 1) {
+		while (this.selected > 0 && this.options[this.selected].isDisabled) {
 			this.selected--;
 		}
```
[the remark this answers is not in the document]
Comment (mrleemurray) [reply src/vs/base/browser/ui/selectBox/selectBoxCustom.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/microsoft/vscode/pull/297703#discussion_r2853773710)

**Review by bpasero — APPROVED** [review]
@mrleemurray some review feedback from AI, but I am not sure how relevant:

1. **Invalid ARIA role on `<option>`** — selectBoxNative.ts sets `role="separator"` on an `<option>` element, which is not a valid ARIA role for that element. Remove it or use a different approach for the native path.

2. **Menu selection colors silently changed** — defaultStyles.ts replaces `menuSelectionForeground`/`menuSelectionBackground` with `listHoverForeground`/`listHoverBackground`. This breaks theme customizations for `menu.selectionForeground` and `menu.selectionBackground`. Should be reverted or split into a separate intentional change.

3. **No tests** — The keyboard navigation refactoring (`onDownArrow`, `onUpArrow`, `onHome`, `onEnd`, `onPageUp`, `onPageDown`) is non-trivial and needs tests, especially for edge cases: all options disabled, separators at boundaries, multiple contiguous separators.

4. **`onPointerUp` doesn't guard against disabled/separator items** — Mouse-over now correctly skips disabled items, but clicking directly on a separator row could still trigger selection. `onPointerUp` needs the same guard.

5. **Separator `::after` needs `position: relative` on parent** — In selectBoxCustom.css, the absolute-positioned `::after` line relies on the list widget setting position on `.monaco-list-row`. Add `position: relative` to `.option-separator` to make it self-contained.

6. **`getComputedStyle` called on every dropdown render** — selectBoxCustom.ts forces a style recalc each time. Use `font-size: inherit` in CSS instead, or cache the value.

7. **`isSeparator` should imply `isDisabled`** — The renderer in selectBoxCustom.ts manually adds `option-disabled` for separators, but this invariant isn't enforced. Either enforce it at the type/setter level, or document it clearly. As-is, a consumer could set `isSeparator: true` without `isDisabled: true` and get inconsistent behavior (CSS shows disabled, but navigation doesn't skip it).

8. **Unrelated visual changes should be split out** — Menu border-radius, item height (`2em` → `24px`), focus outline style (`1.6px dotted` → `1px solid`), dropdown corner radius (`small` → `large`), theme-2026 backdrop-filter additions, and dropdown container outline removal are all unrelated to the separator feature and should be in a separate PR for clean bisection.
